zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,413 @@ where
 
     Ok(())
 }
+
+/// Type alias for a boxed state service.
+type BoxStateService =
+    BoxService<zebra_state::Request, zebra_state::Response, zebra_state::BoxError>;
+
+/// Type alias for the RPC client to communicate with a lightwalletd instance.
+type LightwalletdRpcClient = lightwalletd::rpc::compact_tx_streamer_client::CompactTxStreamerClient<
+    tonic::transport::Channel,
+>;
+
+/// Optional environment variable with the cached state for lightwalletd.
+///
+/// Can be used to speed up the [`sending_transactions_using_lightwalletd`] test, by allowing the
+/// test to reuse the cached lightwalletd synchronization data.
+const LIGHTWALLETD_DATA_DIR_VAR: &str = "LIGHTWALLETD_DATA_DIR";
+
+/// The maximum time to wait for Zebrad to synchronize up to the chain tip starting from a
+/// partially synchronized state.
+///
+/// The partially synchronized state is expected to be close to the tip, so this timeout can be
+/// lower than what's expected for a full synchronization. However, a value that's too short may
+/// cause the test to fail.
+const FINISH_PARTIAL_SYNC_TIMEOUT: Duration = Duration::from_secs(60 * 60);
+
+/// The maximum time that a `lightwalletd` integration test is expected to run.
+const LIGHTWALLETD_TEST_TIMEOUT: Duration = Duration::from_secs(60 * 60);
+
+/// Test sending transactions using a lightwalletd instance connected to a zebrad instance.
+///
+/// This test requires a cached chain state that is partially synchronized, i.e., it should be a
+/// few blocks below the network chain tip height.
+///
+/// The transactions to use to send are obtained from the blocks synchronized by a temporary zebrad
+/// instance that are higher than the chain tip of the cached state.
+///
+/// The zebrad instance connected to lightwalletd uses the cached state and does not connect to any
+/// external peers, which prevents it from downloading the blocks from where the test transactions
+/// were obtained. This is to ensure that zebra does not reject the transactions because they have
+/// already been seen in a block.
+#[tokio::test]
+async fn sending_transactions_using_lightwalletd() -> Result<()> {
+    zebra_test::init();
+
```

Comment:
Blocker: Can we skip this test unless it is specifically requested?

Here's what we do in `lightwalletd_integration()`:
```suggestion

    // Skip the test unless the user specifically asked for it
    if zebra_skip_lightwalletd_tests() {
        return Ok(());
    }

```

Reply:
Done.